server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -958,6 +1051,27 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
             if (format != null) {
                 throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support formats.");
             }
+            if (this.script != null) {
+                // values don't live in source - either pull from dv, or re-calculate
+                if (hasDocValues()) {
+                    return new DocValueFetcher(DocValueFormat.RAW, context.getForField(this));
```

Comment:
It'd be good to discuss the upsides/ downsides of this approach. We've warned users against loading too many doc value fields, since each one involves a seek. We enforce a soft limit of 100 doc value fields through the `index.max_docvalue_fields_search` setting. In a way, defaulting to doc values here bypasses that limit. I realize that runtime fields can already fetch through doc values, but maybe we have different performance/ consistency standards for calculated fields vs. runtime fields?

Reply:
To be consistent with how things are done for non-scripted values, I think we should remove the docvalues branch here.  It would be nice to be able to provide a hint at fetch time as to what the best way of loading values is (ie use docvalues if you're only loading a few values and aren't loading source, etc) but for now let's keep it simple.

Reply:
that means that we always recalculate, a bit like the fields API always fetches from _source?

Reply:
Yes. I think that's ok though, these are only used at fetch time.

Reply:
I agree